dace/transformation/transformation.py

```diff
@@ -296,11 +378,65 @@ def from_json(json_obj, context=None):
         return ret
 
 
+class PatternNode(object):
+    """ 
+    Static field wrapper of a node or an SDFG state that designates it as part
+    of a subgraph pattern. These objects are used in subclasses of 
+    `Transformation` to represent the subgraph patterns.
+
+    Example use:
+    ```
+    @registry.autoregister_params(singlestate=True)
+    class MyTransformation(Transformation):
+        some_map_node = PatternNode(nodes.MapEntry)
+        array = PatternNode(nodes.AccessNode)
+    ```
+
+    The two nodes can then be used in the transformation static methods (e.g.,
+    `expressions`, `can_be_applied`) to represent the nodes, and in the instance
+    methods to point to the nodes in the parent SDFG.
+    """
+    def __init__(self, nodeclass: Type[Union[nd.Node, SDFGState]]) -> None:
```

Comment:
can we avoid the union of types? For example with two classmethods `instance_node(node: nd.Node)`, `instance_state(state: SDFGState)` that returns instance of PatternNode?

Reply:
That would make the syntax more confusing rather than what it looks like now (`PatternNode(class)`). If you want, I can just make it Type and it will be completely unclear what it might be.

Reply:
Just Type or Any is definitely worse. I don't see why it will be more confusing with two classmethods. My main argument is that if you see call instance_node(x) somewhere in the code, you immediately understand what is the type of x. With the current interface, I would need to run a debugger to see what exactly is passed.